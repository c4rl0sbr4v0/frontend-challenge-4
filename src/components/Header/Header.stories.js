import React from 'react';
import Header from './index';

export default {
  title: 'Components/Header',
  component: Header,
};

export const Default = () => (
  <header class="hero is-dark">
    <Header>Landbot</Header>
  </header>
);
