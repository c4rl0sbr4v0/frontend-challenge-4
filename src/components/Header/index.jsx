import React from 'react';

/**
 *  Header: Render a Header with content provided
 * @param {Object} props
 */
const Header = ({ children }) => <div className="landbot-header">{children}</div>;

export default React.memo(Header, () => true);
