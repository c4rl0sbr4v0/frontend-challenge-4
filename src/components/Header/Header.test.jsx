import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '.';

test('Header Component', async () => {
  const { container } = render(<Header>header</Header>);
  expect(container.firstChild.className === 'landbot-header').toBe(true);
  expect(container).toMatchSnapshot();
});
