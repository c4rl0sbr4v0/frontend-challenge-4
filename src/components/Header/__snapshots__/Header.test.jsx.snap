// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component 1`] = `
<div>
  <div
    class="landbot-header"
  >
    header
  </div>
</div>
`;
